packages/dds/tree/src/test/shared-tree/sharedTreeFuzzTests.ts

```diff
@@ -0,0 +1,287 @@
+                const parentIndex = contents.parent?.parentIndex;
+                if (delField !== undefined && parent !== undefined && parentIndex !== undefined) {
+                    tree.runTransaction((forest, editor) => {
+                        const field = editor.sequenceField(parent, delField);
+                        field.delete(parentIndex, 1);
+                        return transactionResult;
+                    });
+                }
+            }
+            break;
+        case "setPayload":
+            if (index !== undefined && nodeField !== undefined) {
+                const path = contents.parent;
+                if (path !== undefined) {
+                    tree.runTransaction((forest, editor) => {
+                        editor.setValue(path, contents.value);
+                        return transactionResult;
+                    });
+                }
+            }
+            break;
+        default:
+            fail("Invalid edit.");
+    }
+}
+
+enum FuzzTestType {
+    Basic = 0,
+    Abort = 1,
+    AnchorStability = 2,
+}
+
+export function runSharedTreeFuzzTests(title: string): void {
+    const random = makeRandom(0)
+    const testBatchSize = 3;
+    describeFuzz(title, ({ testCount }) => {
+        function runFuzzTest (
+            opGenerator: () => AsyncGenerator<Operation, FuzzTestState>,
+            fuzzTestType: FuzzTestType,
+            stepSize: number,
+            batchSize: number
+        ): void {
+            const seed = random.integer(1, 1000000);
+            for (let i = 0; i < batchSize; i++) {
+                const innerRandom = makeRandom(seed);
+                const generatorFactory = () => take(stepSize, opGenerator());
+                switch (fuzzTestType) {
+                    case FuzzTestType.Basic:
+                        it(`with seed ${i}`, async () => {
+                            await performFuzzActions(generatorFactory(), innerRandom.integer(1, 1000000));
+                        }).timeout(20000);
+                        break;
+                    case FuzzTestType.Abort:
+                        it.skip(`with seed ${i}`, async () => {
+                            await performFuzzActionsAbort(generatorFactory(), innerRandom.integer(1, 1000000));
+                        }).timeout(20000);
+                        break;
+                    case FuzzTestType.AnchorStability:
+                        it.skip(`with seed ${i}`, async () => {
+                            await performFuzzActionsAbort(generatorFactory(), innerRandom.integer(1, 1000000), true);
+                        }).timeout(20000);
+                        break;
+                    default:
+                        fail("Invalid FuzzTestType.");
+                }
+            }
+        }
+        describe("basic convergence", () => {
+            describe("using TestTreeProvider", () => {
+                for (let stepSize = 0; stepSize < 1000; stepSize += 100){
```

Comment:
I don't think we want to run the tests with different step sizes.
I'd expect the sweet spot to be at about 20 steps. The idea is that we want enough steps to make all relevant scenarios possible, but we want as few steps as possible beyond that because debugging scenarios that fail on later steps will be extra annoying.